### PR TITLE
Functional tx test type 1 - Closes #824

### LINKED ISSUE #824: Functional tx test type 1 - Subtask 678
Several actions are taken to introduce the second file for funcional post transactions:

- review GET `accounts.js`
- introduce POST `1.second.secret.js` : this file is quite complex due to the introduction of *enforcement* phase. It checks if the second secret freshly settled is asked when sending any kind of transaction.

Subtask of #678

### PR DESCRIPTION
**Pull request by diego-G**
Closes #824

### CODE REVIEW

File: test/functional/http/get/accounts.js [hunk 1]
```diff
@@ -1,268 +1,239 @@
...
+				node.expect(res.account).to.have.property('multisignatures').to.a('array');
+				node.expect(res.account).to.have.property('u_multisignatures').to.a('array');
+			});
+		});
+
+		it('using known lowercase address should be ok', function () {
+			return getAccountsPromise('address=' + node.gAccount.address.toLowerCase()).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('account').that.is.an('object');
+				node.expect(res.account).to.have.property('address').to.equal(node.gAccount.address);
```
Comment (4miners) [inline test/functional/http/get/accounts.js]: We expect here address with uppercase `L` - that should be noticed somewhere.

File: test/functional/http/post/1.second.secret.js [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+			return onNewBlockPromise();
+		});
+	});
+
+	describe('schema validations', function () {
+
+		shared.invalidAssets(account, 'signature', badTransactions);
+	});
+
+	describe('transactions processing', function () {
```
Comment (4miners) [inline test/functional/http/post/1.second.secret.js]: not very clear description

File: test/functional/http/post/1.second.secret.js [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+			transaction = node.lisk.signature.createSignature(accountNoFunds.password, accountNoFunds.secondPassword);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.not.be.ok;
+				node.expect(res).to.have.property('message').to.equal('Account does not have enough LSK: ' + accountNoFunds.address + ' balance: 0');
+				badTransactions.push(transaction);
+			});
+		});
+
+		it('with exact funds should be ok', function () {
```
Comment (4miners) [inline test/functional/http/post/1.second.secret.js]: same here, what `exact funds` mean?

File: test/functional/http/post/1.second.secret.js [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+			transaction = node.lisk.signature.createSignature(accountEmptySecondPassword.password, accountEmptySecondPassword.secondPassword);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				goodTransactions.push(transaction);
+			});
+		});
+
+		it('with valid params should be ok but not confirmed if sendind twice in a row', function () {
```
Comment (4miners) [inline test/functional/http/post/1.second.secret.js]: typo

File: test/functional/http/post/1.second.secret.js [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				goodTransactions.push(transaction);
+			});
+		});
+	});
+
+	describe('enforcement before confirmation', function () {
```
Comment (4miners) [inline test/functional/http/post/1.second.secret.js]: what?

File: test/functional/http/post/1.second.secret.js [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+			});
+		});
+	});	
+
+	describe('transactions confirmation', function () {
+
+		shared.confirmationPhase(goodTransactions, badTransactions);
+	});
+
+	describe('enforcement', function () {
```
Comment (4miners) [inline test/functional/http/post/1.second.secret.js]: Again, idk about what those tests are about

File: test/functional/http/get/accounts.js [hunk 7]
```diff
@@ -1,268 +1,239 @@
...
+
+		it('using unknown address should fail', function () {
+			return getAccountsPromise('address=' + account.address).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.not.ok;
+				node.expect(res).to.have.property('error').to.eql('Account not found');
+			});
+		});
+
+		it('using invalid address should fail', function () {
+			return getAccountsPromise('address=' + 'thisIsNOTALiskAddress').then(function (res) {
```
Comment (MaciejBaj) [inline test/functional/http/get/accounts.js]: Maybe just rename to `invalidAddress` as in different tests

File: test/functional/http/post/1.second.secret.js [hunk 8]
```diff
@@ -0,0 +1,456 @@
...
+
+var sendTransactionPromise = require('../../../common/apiHelpers').sendTransactionPromise;
+var creditAccountPromise = require('../../../common/apiHelpers').creditAccountPromise;
+var sendSignaturePromise = require('../../../common/apiHelpers').sendSignaturePromise;
+
+var onNewBlockPromise = node.Promise.promisify(node.onNewBlock);
+
+describe('POST /api/transactions (type 1) register second secret', function () {
+
+	var badTransactions = [];
```
Comment (MaciejBaj) [inline test/functional/http/post/1.second.secret.js]: valid & invalid

File: test/functional/http/post/1.second.secret.js [hunk 9]
```diff
@@ -0,0 +1,456 @@
...
+
+				it('with not all the signatures should be ok but never confirmed', function () {
+					signature = node.lisk.multisignature.signTransaction(pendingMultisignatures[0], accountNoFunds.password);
+
+					return sendSignaturePromise(signature, pendingMultisignatures[0]).then(function (res) {
+						node.expect(res).to.have.property('success').to.be.ok;
+					});
+				});
+
+				// FIXME: affect severily when registering dapp with the same account
```
Comment (MaciejBaj) [inline test/functional/http/post/1.second.secret.js]: Type: `severely`

File: test/functional/http/post/1.second.secret.js (hunk 8 above)
Comment (diego-G) [reply test/functional/http/post/1.second.secret.js]: It is already in place on `0.transfer.js`. We agreed in the past.

File: test/functional/shared.js [hunk 10]
```diff
@@ -167,6 +167,16 @@ function invalidAssets (account, option, badTransactions) {
...
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.not.ok;
+				node.expect(res).to.have.property('message').that.is.not.empty;
+				badTransactions.push(transaction);
+			});
+		});
 	});
 
 	describe('using invalid asset.' + option + ' values', function () {
```
[the remark this answers is not in the document]
Comment (diego-G) [reply test/functional/shared.js]: It should. I aim to use it for `asset.signature`, `asset.delegate`, etc.

File: test/functional/http/post/1.second.secret.js [hunk 11]
```diff
@@ -0,0 +1,438 @@
...
+			transaction = node.lisk.signature.createSignature(accountNoFunds.password, accountNoFunds.secondPassword);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.not.be.ok;
+				node.expect(res).to.have.property('message').to.equal('Account does not have enough LSK: ' + accountNoFunds.address + ' balance: 0');
+				badTransactions.push(transaction);
+			});
+		});
+
+		it('with scarce funds should be ok', function () {
```
Comment (4miners) [inline test/functional/http/post/1.second.secret.js]: `using all funds should be ok`

Comment (diego-G) [reply test/functional/http/post/1.second.secret.js]: When I define the case, I credit the account with `scarce funds` meaning short supply of funds which are going to be spent.

File: test/functional/shared.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (Isabello) [reply test/functional/shared.js]: ok.

File: test/functional/http/post/1.second.secret.js [hunk 12]
```diff
@@ -0,0 +1,438 @@
...
+			transaction = node.lisk.signature.createSignature(accountEmptySecondPassword.password, accountEmptySecondPassword.secondPassword);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				goodTransactions.push(transaction);
+			});
+		});
+
+		it('with valid params should be ok but not confirmed if sending twice in a row', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: with valid params and duplicate submission should be ok

File: test/functional/http/post/1.second.secret.js [hunk 13]
```diff
@@ -0,0 +1,438 @@
...
+				goodTransactions.push(transaction);
+			});
+		});
+	});
+
+	describe('unconfirmed state', function () {
+
+		describe('type 0 - sending funds', function () {
+
+			it('using correct second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using a second passphrase on an account without second passphrase should fail

File: test/functional/http/post/1.second.secret.js [hunk 14]
```diff
@@ -0,0 +1,438 @@
...
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Sender does not have a second signature');
+					badTransactions.push(transaction);
+				});
+			});
+		});
+
+		describe('type 2 - registering delegate', function () {
+
+			it('using correct second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: again there can be no correct second passphrase on an account without one.

File: test/functional/http/post/1.second.secret.js [hunk 15]
```diff
@@ -0,0 +1,438 @@
...
+	});	
+
+	describe('confirmation', function () {
+
+		shared.confirmationPhase(goodTransactions, badTransactions);
+	});
+
+	describe('validation', function () {
+
+		it('using second passphrase on a fresh account should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: is this not already tested by the previous group of tests?

File: test/functional/http/post/1.second.secret.js [hunk 16]
```diff
@@ -0,0 +1,438 @@
...
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.not.ok;
+				node.expect(res).to.have.property('message').to.equal('Sender does not have a second signature');
+				badTransactionsEnforcement.push(transaction);
+			});
+		});
+
+		describe('type 0 - sending funds', function () {
+
+			it('using no second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using no second passphrase on account with second passphrase enabled should fail

File: test/functional/http/post/1.second.secret.js [hunk 17]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, account.password);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using invalid second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using second passphrase that does not match registered secondPublicKey should fail

File: test/functional/http/post/1.second.secret.js [hunk 18]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.transaction.createTransaction(node.eAccount.address, 1, account.password, account.secondPassword);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using correct empty second passphrase should be ok', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: Irrelevant test. We dont need to test empty second passphrases.

File: test/functional/http/post/1.second.secret.js [hunk 19]
```diff
@@ -0,0 +1,438 @@
...
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+		});
+
+		describe('type 1 - second secret', function () {
+
+			it('setting second signature twice on the same account should be not ok', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: registering a second passphrase on an account with a second passphrase should fail

File: test/functional/http/post/1.second.secret.js [hunk 20]
```diff
@@ -0,0 +1,438 @@
...
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+		});
+
+		describe('type 2 - registering delegate', function () {
+
+			it('using no second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using no second passphrase with account that has a second passphrase registered should fail

File: test/functional/http/post/1.second.secret.js [hunk 21]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.delegate.createDelegate(account.password, account.username);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using invalid second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using second passphrase that does not match registered secondPublicKey should fail

File: test/functional/http/post/1.second.secret.js [hunk 22]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.delegate.createDelegate(account.password, account.username, account.secondPassword);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using correct empty second passphrase should be ok', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: Should not bother testing, empty string is as good as a full string when it comes to passphrases.

File: test/functional/http/post/1.second.secret.js [hunk 23]
```diff
@@ -0,0 +1,438 @@
...
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+		});
+
+		describe('type 3 - voting delegate', function () {
+
+			it('using no second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using no second passphrase with account that has a second passphrase registered should fail

File: test/functional/http/post/1.second.secret.js [hunk 24]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.vote.createVote(account.password, ['+' + node.eAccount.publicKey]);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using invalid second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using second passphrase that does not match registered secondPublicKey should fail

File: test/functional/http/post/1.second.secret.js [hunk 25]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.vote.createVote(account.password, ['+' + node.eAccount.publicKey], account.secondPassword);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using correct empty second passphrase should be ok', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: same as the other comments

File: test/functional/http/post/1.second.secret.js [hunk 26]
```diff
@@ -0,0 +1,438 @@
...
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactionsEnforcement.push(transaction);
+				});
+			});
+		});
+
+		describe('type 4 - registering multisignature account', function () {
+
+			it('using no second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using no second passphrase with account that has a second passphrase registered should fail

File: test/functional/http/post/1.second.secret.js [hunk 27]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.multisignature.createMultisignature(account.password, null, ['+' + node.eAccount.publicKey, '+' + accountNoFunds.publicKey, '+' + accountMinimalFunds.publicKey], 1, 2);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using invalid second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using second passphrase that does not match registered secondPublicKey should fail

File: test/functional/http/post/1.second.secret.js [hunk 28]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.multisignature.createMultisignature(account.password, 'wrong second password', ['+' + node.eAccount.publicKey, '+' + accountNoFunds.publicKey, '+' + accountMinimalFunds.publicKey], 1, 2);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Failed to verify second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using correct empty second passphrase should be ok', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: irrelevant, should have the same order as the other types regardless.

File: test/functional/http/post/1.second.secret.js [hunk 29]
```diff
@@ -0,0 +1,438 @@
...
+					return sendSignaturePromise(signature, pendingMultisignatures[0]).then(function (res) {
+						node.expect(res).to.have.property('success').to.be.ok;
+					});
+				});
+			});
+		});
+
+		describe('type 5 - registering dapp', function () {
+
+			it('using no second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: as in the other comments

File: test/functional/http/post/1.second.secret.js [hunk 30]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.dapp.createDapp(account.password, null, node.randomApplication());
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using invalid second passphrase should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: as in the other comments

File: test/functional/http/post/1.second.secret.js [hunk 31]
```diff
@@ -0,0 +1,438 @@
...
+				transaction = node.lisk.dapp.createDapp(account.password, 'wrong second password', node.randomApplication());
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Failed to verify second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+
+			it('using correct empty second passphrase should be ok', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: as in the other comments

File: test/functional/http/post/1.second.secret.js [hunk 32]
```diff
@@ -0,0 +1,388 @@
...
+				goodTransactions.push(transaction);
+			});
+		});
+	});
+
+	describe('unconfirmed state', function () {
+
+		describe('type 0 - sending funds', function () {
+
+			it('using a second passphrase on an account wih second passphrase in process should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: using second signature with an account that has a pending second passphrase registration should fail

File: test/functional/http/post/1.second.secret.js [hunk 33]
```diff
@@ -0,0 +1,388 @@
...
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Missing sender second signature');
+					badTransactionsEnforcement.push(transaction);
+				});
+			});
+		});
+
+		describe('type 2 - registering delegate', function () {
+
+			it('using no second passphrase on an account with second pasphrase enabled should fail', function () {
```
Comment (Isabello) [inline test/functional/http/post/1.second.secret.js]: passphrase
